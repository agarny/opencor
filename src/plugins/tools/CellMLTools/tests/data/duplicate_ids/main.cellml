<?xml version='1.0' encoding='UTF-8'?>
<model name="test_1_1_model" xmlns="http://www.cellml.org/cellml/1.1#" xmlns:cellml="http://www.cellml.org/cellml/1.1#" xmlns:xlink="http://www.w3.org/1999/xlink" cmeta:id="test_1_1_model" xmlns:cmeta="http://www.cellml.org/metadata/1.0#">
    <import xlink:href="sub_model.cellml">
        <component component_ref="main" name="sub_model_main"/>
    </import>
    <import xlink:href="other_model.cellml">
        <component component_ref="main" name="other"/>
    </import>
    <component name="main" cmeta:id="qwe" xmlns:cmeta="http://www.cellml.org/metadata/1.0#">
        <variable name="result" units="metre" cmeta:id="qwe1" xmlns:cmeta="http://www.cellml.org/metadata/1.0#"/>
        <variable name="in1" public_interface="in" units="metre" cmeta:id="qwe2" xmlns:cmeta="http://www.cellml.org/metadata/1.0#"/>
        <variable name="in2" public_interface="in" units="metre" cmeta:id="qwe3" xmlns:cmeta="http://www.cellml.org/metadata/1.0#"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML">
            <apply>
                <eq/>
                <ci>result</ci>
                <apply>
                    <plus/>
                    <ci>in1</ci>
                    <ci>in2</ci>
                    <cn cellml:units="metre">1</cn>
                </apply>
            </apply>
        </math>
    </component>
    <connection>
        <map_components component_1="main" component_2="sub_model_main"/>
        <map_variables variable_1="in1" variable_2="output"/>
    </connection>
    <connection>
        <map_components component_1="main" component_2="other"/>
        <map_variables variable_1="in2" variable_2="output"/>
    </connection>
    <connection>
        <map_components component_1="other" component_2="sub_model_main"/>
        <map_variables variable_1="input" variable_2="side_out"/>
    </connection>
</model>
