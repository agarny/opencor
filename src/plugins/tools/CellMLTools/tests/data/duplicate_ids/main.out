<?xml version='1.0' encoding='UTF-8'?>
<model cmeta:id="test_1_1_model" name="test_1_1_model" xmlns="http://www.cellml.org/cellml/1.0#" xmlns:cmeta="http://www.cellml.org/metadata/1.0#">
    <component cmeta:id="qwe" name="main">
        <variable cmeta:id="qwe1" name="result" units="metre"/>
        <variable cmeta:id="qwe2" name="in1" public_interface="in" units="metre"/>
        <variable cmeta:id="qwe3" name="in2" public_interface="in" units="metre"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>result</ci>
                <apply>
                    <plus/>
                    <ci>in1</ci>
                    <ci>in2</ci>
                    <cn cellml:units="metre">1</cn>
                </apply>
            </apply>
        </math>
    </component>
    <component cmeta:id="b4da55" name="sub_model_main">
        <variable cmeta:id="b4da56" name="output" public_interface="out" units="metre"/>
        <variable cmeta:id="b4da57" name="side_out" private_interface="in" public_interface="out" units="dimensionless"/>
        <variable cmeta:id="b4da58" name="input" private_interface="in" units="metre"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>output</ci>
                <apply>
                    <times/>
                    <ci>input</ci>
                    <cn cellml:units="dimensionless">2</cn>
                </apply>
            </apply>
        </math>
    </component>
    <component cmeta:id="b4da59" name="other">
        <variable cmeta:id="b4da5a" name="output" public_interface="out" units="metre"/>
        <variable cmeta:id="b4da5b" name="input" public_interface="in" units="dimensionless"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>output</ci>
                <apply>
                    <times/>
                    <ci>input</ci>
                    <cn cellml:units="metre">1</cn>
                </apply>
            </apply>
        </math>
    </component>
    <component cmeta:id="b4da5c" name="sub_sub_model">
        <variable cmeta:id="b4da5d" name="output" public_interface="out" units="metre"/>
        <variable cmeta:id="b4da5e" name="input" private_interface="in" units="metre"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>output</ci>
                <apply>
                    <divide/>
                    <ci>input</ci>
                    <cn cellml:units="dimensionless">2</cn>
                </apply>
            </apply>
        </math>
    </component>
    <component cmeta:id="asd" name="side">
        <variable cmeta:id="asd1" name="output" public_interface="out" units="dimensionless"/>
        <variable cmeta:id="asd2" name="input" private_interface="in" units="dimensionless"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>output</ci>
                <apply>
                    <plus/>
                    <ci>input</ci>
                    <cn cellml:units="dimensionless">5</cn>
                </apply>
            </apply>
        </math>
    </component>
    <component cmeta:id="zxc" name="inside">
        <variable cmeta:id="zxc1" name="output" public_interface="out" units="dimensionless"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>output</ci>
                <apply>
                    <plus/>
                    <cn cellml:units="dimensionless">1</cn>
                    <cn cellml:units="dimensionless">2</cn>
                    <cn cellml:units="dimensionless">3</cn>
                    <cn cellml:units="dimensionless">4</cn>
                </apply>
            </apply>
        </math>
    </component>
    <component cmeta:id="b4da5f" name="sub">
        <variable cmeta:id="b4da60" name="output" public_interface="out" units="metre"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
            <apply>
                <eq/>
                <ci>output</ci>
                <apply>
                    <minus/>
                    <cn cellml:units="metre">20</cn>
                </apply>
            </apply>
        </math>
    </component>
    <group>
        <relationship_ref relationship="encapsulation"/>
        <component_ref component="sub_model_main">
            <component_ref component="sub_sub_model"/>
            <component_ref component="side">
                <component_ref component="inside"/>
            </component_ref>
        </component_ref>
        <component_ref component="sub_sub_model">
            <component_ref component="sub"/>
        </component_ref>
    </group>
    <connection>
        <map_components component_1="main" component_2="sub_model_main"/>
        <map_variables variable_1="in1" variable_2="output"/>
    </connection>
    <connection>
        <map_components component_1="main" component_2="other"/>
        <map_variables variable_1="in2" variable_2="output"/>
    </connection>
    <connection>
        <map_components component_1="other" component_2="sub_model_main"/>
        <map_variables variable_1="input" variable_2="side_out"/>
    </connection>
    <connection>
        <map_components component_1="sub_model_main" component_2="sub_sub_model"/>
        <map_variables variable_1="input" variable_2="output"/>
    </connection>
    <connection>
        <map_components component_1="sub_model_main" component_2="side"/>
        <map_variables variable_1="side_out" variable_2="output"/>
    </connection>
    <connection>
        <map_components component_1="inside" component_2="side"/>
        <map_variables variable_1="output" variable_2="input"/>
    </connection>
    <connection>
        <map_components component_1="sub_sub_model" component_2="sub"/>
        <map_variables variable_1="input" variable_2="output"/>
    </connection>
</model>
